bin.js

```diff
@@ -4,13 +4,22 @@ var proc = require('child_process')
 var os = require('os')
 var path = require('path')
 
+var targetMismatch = (process.env.npm_config_platform && process.platform !== process.env.npm_config_platform) ||
+  (process.env.npm_config_arch && process.arch !== process.env.npm_config_arch)
+
 if (!buildFromSource()) {
-  proc.exec('node-gyp-build-test', function (err, stdout, stderr) {
-    if (err) {
-      if (verbose()) console.error(stderr)
-      preinstall()
-    }
-  })
+  if (targetMismatch) {
+    // If the target is mismatched, then we can't test built output, and we can't build from source either.
+    // Instead, we just check whether a matching prebuild exists, and fail if it doesn't.
+    require('./index').path(process.cwd())
+  } else {
+    proc.exec('node-gyp-build-test', function (err, stdout, stderr) {
```

Comment:
@mafintosh Do you want a custom test script (https://github.com/prebuild/node-gyp-build/commit/f1c799655b867efc4802018a10cfb7c9d8c2ea8a) to be executed even if platform or arch don't match?

Reply:
no, don't think that's needed